src/znc.cpp

```diff
@@ -994,8 +994,43 @@ void CZNC::BackupConfigOnce(const CString& sSuffix) {
         CUtils::PrintStatus(false, strerror(errno));
 }
 
-bool CZNC::ParseConfig(const CString& sConfig, CString& sError) {
+bool CZNC::AcquireLock(CString& sError) {
+    // If we already have the lock file, do nothing
+    // FIXME: check if we actually hold the fcntl lock
+    if (m_pLockFile) return true;
+
+    CFile* pLock = new CFile(m_sConfigFile + ".lock");
+    // need to open the lock file Read/Write for fcntl()
+    // exclusive locking to work properly!
+    if (!pLock->Open(O_RDWR|O_CREAT, 0600)) {
+        sError = "Can not open lock file";
+        CUtils::PrintStatus(false, sError);
+        delete pLock;
+        return false;
+    }
+
+    if (!pLock->TryExLock()) {
+        sError = "ZNC is already running on this config.";
+        CUtils::PrintStatus(false, sError);
+        delete pLock;
+        return false;
+    }
+
+    m_pLockFile = pLock;
+    return true;
+}
+
+bool CZNC::ParseConfig(const CString& sConfig, bool bReadonlyConfig, CString& sError) {
     m_sConfigFile = ExpandConfigPath(sConfig, false);
+    m_bReadonlyConfig = bReadonlyConfig;
+
+    if (m_bReadonlyConfig) {
+        CUtils::PrintError("Running in read-only mode. Any changes to the settings, users,");
```

Comment:
Why is it an error?

Reply:
I wanted it to be a warning, but couldn't find anything like `PrintWarning`. I looked at what `--allow-root` does, and it prints its warning as an error.